api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,16 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						// if 'name' is a file then make sure that only path is trimmed out and not the file name itself 
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
I'm trying to go by this saying [don't try to fix it if ain't broke](http://www.phrases.org.uk/meanings/if-it-aint-broke-dont-fix-it.html), unless something is failing because this is missing then we will fix it at that point, as long as all reference to it in core set the parameters to true and the file name respectively it all good the way it is otherwise you risk introducing other obscure bugs, we don't expect external code to be calling this method.